core/trino-parser/src/main/java/io/trino/sql/QueryUtil.java

```diff
@@ -267,6 +268,24 @@ public static Query singleValueQuery(String columnName, boolean value)
                 aliased(values, "t", ImmutableList.of(columnName)));
     }
 
+    public static Query emptyQuery(List<String> columns)
+    {
+        Select select = selectList(columns.stream()
+                .map(column -> new SingleColumn(new NullLiteral(), QueryUtil.identifier(column)))
```

Comment:
as a follow up we might want to have the columns be of proper type, eg varchar for role names

Reply:
consider adding todo here